File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int r = matrix.size();
        int c = matrix[0].size();

        bool r0 = false;
        bool c0 = false;
        bool temp = false;

        if(matrix[0][0] == 0) temp = true;

        for(int i=1; i<r; i++){
            if(matrix[i][0] == 0){
                c0 = true;
                break;
            }
        }

        for(int j=1; j<c; j++){
            if(matrix[0][j] == 0){
                r0 = true;
                break;
            }
        }

        for(int i=1; i<r; i++){
            for(int j=1; j<c; j++){
                if(matrix[i][j] == 0){
                    matrix[i][0] = 0;
                    matrix[0][j] = 0;
                }
            }
        }

        for(int i=1; i<r; i++){
            if(matrix[i][0] == 0){
                for(int j=1; j<c; j++){
                    matrix[i][j] = 0;
                }
            }
        }

        for(int i=1; i<c; i++){
            if(matrix[0][i] == 0){
                for(int j=1; j<r; j++){
                    matrix[j][i] = 0;
                }
            }
        }

        if(r0){
            for(int i=0; i<c; i++){
                matrix[0][i] = 0;
            }
        }

        if(c0){
            for(int j=0; j<r; j++){
                matrix[j][0] = 0;
            }
        }

        if(temp){
            for(int i=0; i<r; i++){
                matrix[i][0] = 0;
            }

            for(int i=0; i<c; i++){
                matrix[0][i] = 0;
            }
        }
    }
};

File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:

        r = len(matrix)
        c = len(matrix[0])

        r0 = False
        c0 = False
        temp = False

        if(matrix[0][0] == 0):
            temp = True

        for i in range(1, r):
            if(matrix[i][0] == 0):
                c0 = True
                break
        
        for i in range(1, c):
            if(matrix[0][i] == 0):
                r0 = True
                break

        for i in range(1,r):
            for j in range(1,c):
                if(matrix[i][j] == 0):
                    matrix[i][0] = 0
                    matrix[0][j] = 0

        for i in range(1,r):
            if(matrix[i][0] == 0):
                for j in range(1,c):
                    matrix[i][j] = 0

        for i in range(1,c):
            if(matrix[0][i] == 0):
                for j in range(1,r):
                    matrix[j][i] = 0

        if(c0):
            for i in range(r):
                matrix[i][0] = 0
            
        if(r0):
            for i in range(c):
                matrix[0][i] = 0
        
        if(temp):
            for i in range(r):
                matrix[i][0] = 0
            
            for i in range(c):
                matrix[0][i] = 0
        